R/integrations.R

```diff
@@ -161,3 +161,115 @@ basicSeuratProc <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, do
+  cms_list = lapply(cms_list, function(x) {lapply(x, function(y) {y[row.names(y) %in% common_genes,]} )} )
+  
+  # Merge velocity files from different samples
+  emat <- do.call(cbind, lapply(cms_list, function(x) {x[[1]]}))
+  nmat <- do.call(cbind, lapply(cms_list, function(x) {x[[2]]}))
+  smat <- do.call(cbind, lapply(cms_list, function(x) {x[[3]]}))
+  
+  cluster.label <- con$clusters$leiden$groups
+  cell.colors <- pagoda2:::fac2col(cluster.label)
+  emb <- con$embedding
+  
+  # Keep the order of cells consistent between velocity matrices and the embedding (not really sure whether it's necessary...)
+  emat <- emat[,order(match(colnames(emat), rownames(emb)))]
+  nmat <- nmat[,order(match(colnames(nmat), rownames(emb)))]
+  smat <- smat[,order(match(colnames(smat), rownames(emb)))]
+  
+  if (verbose) cat("Calculating cell distances...\n")
+  # Get a pca results for all the samples from the conos object
+  pcs <- PCAforVelo(con$samples, n.odgenes = n.odgenes)
+  # Again, keep the order of cells consistent
+  pcs <- pcs[order(match(rownames(pcs), rownames(emb))),]
+  # Calculate the cell distances as in the original function
+  cell.dist <- as.dist(1 - velocyto.R::armaCor(t(pcs)))
+  
+  if (verbose) cat("Filtering velocity...\n")
+  emat %<>% velocyto.R::filter.genes.by.cluster.expression(cluster.label, min.max.cluster.average=min.max.cluster.average.emat)
+  nmat %<>% velocyto.R::filter.genes.by.cluster.expression(cluster.label, min.max.cluster.average=min.max.cluster.average.nmat)
+  smat %<>% velocyto.R::filter.genes.by.cluster.expression(cluster.label, min.max.cluster.average=min.max.cluster.average.smat)
+  
+  if (verbose) cat("All Done!")
+  return(list(cell.dist=cell.dist, emat=emat, nmat=nmat, smat=smat, cell.colors=cell.colors, emb=emb))
+}
+
+# Intersect genes and cells between all the velocity files and the conos object
+PrepareVelocity <- function(cms_file, genes, cells) {
+  exon_genes <- rownames(cms_file$exon)
+  intron_genes <- rownames(cms_file$intron)
+  spanning_genes <- rownames(cms_file$spanning)
+  # Only common genes between the 3 files and conos object
+  common_genes <- intersect(exon_genes, intron_genes) %>% intersect(spanning_genes) %>% intersect(genes)
+  
+  exon_cells <- colnames(cms_file$exon)
+  intron_cells <- colnames(cms_file$intron)
+  spanning_cells <- colnames(cms_file$spanning)
+  # Only common cells between the 3 files and conos object
+  common_cells <- intersect(exon_cells, intron_cells) %>% intersect(spanning_cells) %>% intersect(cells)
+  
+  cms_file$exon <- cms_file$exon[common_genes,common_cells]
+  cms_file$intron <- cms_file$intron[common_genes,common_cells]
+  cms_file$spanning <- cms_file$spanning[common_genes,common_cells]
+  return(cms_file)
+}
+
+# Get a pca results for all the samples from the conos object
+# This is a modification of the quickPlainPCA function
+PCAforVelo <- function(r.n,data.type='counts',k=30,ncomps=100,n.odgenes=NULL,verbose=TRUE) {
```

Comment:
- `PCAforVelo` to `pcaForVelo`
- Is `r.n` the list of p2 objects? Better name would help.